src/client/src/orangehrmRecruitmentPlugin/components/CandidateProfile.vue

```diff
@@ -0,0 +1,283 @@
+            >
+              <oxd-input-field
+                v-model="profile.keep"
+                type="checkbox"
+                :label="$t('recruitment.content_to_keep_data')"
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-divider />
+        <required-text></required-text>
+        <oxd-form-actions>
+          <oxd-button display-type="ghost" :label="$t('general.cancel')" />
+          <submit-button :label="$t('general.save')" />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import FullNameInput from '@/orangehrmPimPlugin/components/FullNameInput';
+import {APIService} from '@/core/util/services/api.service';
+import {
+  shouldNotExceedCharLength,
+  required,
+  validDateFormat,
+  validPhoneNumberFormat,
+  validEmailFormat,
+  maxFileSize,
+  validFileTypes,
+} from '@/core/util/validation/rules';
+import VacancyDropdown from '@/orangehrmRecruitmentPlugin/components/VacancyDropdown';
+import FileUploadInput from '@/core/components/inputs/FileUploadInput';
+import DateInput from '@/core/components/inputs/DateInput';
+import {navigate} from '@/core/util/helper/navigation';
+export default {
+  name: 'CandidateProfile',
+  components: {
+    DateInput,
+    'vacancy-dropdown': VacancyDropdown,
+    'file-upload-input': FileUploadInput,
+    'full-name-input': FullNameInput,
+  },
+  props: {
+    candidateId: {
+      type: Number,
+      required: true,
+    },
+  },
+  emits: ['getData'],
+  setup(props) {
+    const http = new APIService(
+      'https://c81c3149-4936-41d9-ab3d-e25f1bff2934.mock.pstmn.io',
+      `/recruitment/candidate/${props.candidateId}`,
+    );
+
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      allowedFileTypes: [
+        'application/pdf',
+        'application/msword',
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+        'text/plain',
+        'application/rtf',
+        'application/vnd.oasis.opendocument.text',
+      ],
+      profile: {
+        firstName: '',
+        middleName: '',
+        lastName: '',
+        email: '',
+        contactNumber: '',
+        oldResume: '',
+        notes: '',
+        keywords: '',
+        newResume: null,
+        vacancy: null,
+        resume: null,
+        method: 'keepCurrent',
+        applicationDate: null,
+        keep: null,
+      },
+      rules: {
+        firstName: [required, shouldNotExceedCharLength(30)],
+        lastName: [required, shouldNotExceedCharLength(30)],
+        middleName: [shouldNotExceedCharLength(30)],
+        email: [required, validEmailFormat, shouldNotExceedCharLength(50)],
+        contactNumber: [validPhoneNumberFormat, shouldNotExceedCharLength(25)],
+        keywords: [shouldNotExceedCharLength(250)],
+        applicationDate: [validDateFormat()],
+        resume: [
+          maxFileSize(1024 * 1024),
```

Comment:
@Super-Chama max file size also better to get from BE config via props.

Reply:
Agreed. can move to controller. @Ajiharan007 can we move both this and file types to controller. you can hardcode at controller level for now

Reply:
done